signer-cloud-server/src/main/java/com/wultra/signercloud/server/restapi/ValidationExceptionHandler.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Signer Cloud
+ * Copyright (C) 2025 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.wultra.signercloud.server.restapi;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+/**
+ * Handler for validation exception producing HTTP 400 response.
+ *
+ * @author Michal Rozehnal, michal.rozehnal@wultra.com
+ */
+@ControllerAdvice
+public class ValidationExceptionHandler {
+    private static final String ERROR_STATUS = "ERROR";
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<ErrorResponse> handleValidationException(MethodArgumentNotValidException ex) {
```

Comment:
I know that this is a pattern in our other components, but I am not quite happy with that. It should work without duplicity (and duplicity tends to errors).
> If empty, will default to any exceptions listed in the method argument list.
```suggestion
    @ExceptionHandler
    public ResponseEntity<ErrorResponse> handleValidationException(final MethodArgumentNotValidException ex) {
```

Reply:
Very good observation. I am also agains the duplicity if it is really not needed. Updated